src/controllers/legacycontrollermapping.h

```diff
@@ -1,14 +1,26 @@
 #pragma once
 
+#include <util/assert.h>
+
 #include <QDebug>
 #include <QDir>
+#include <QDomElement>
 #include <QHash>
 #include <QList>
 #include <QSharedPointer>
 #include <QString>
+#include <QtGlobal>
 #include <memory>
+#include <vector>
 
+#include "controllers/legacycontrollersettings.h"
+#include "controllers/legacycontrollersettingslayout.h"
 #include "defs_urls.h"
+#include "preferences/usersettings.h"
+
+#define CONTROLLER_SETTINGS_PREFERENCE_GROUP_KEY "[ControllerSettings_%1_%2]"
+#define CONTROLLER_SETTINGS_SETTING_PATH_SUBST "%SETTING_PATH"
+#define CONTROLLER_SETTINGS_RESOURCE_PATH_SUBST "%RESOURCE_PATH"
```

Comment:
Both can be moved to the anonymous namespace of the CPP file. and can become a `const QString = QStringLiteral ` with k prefix and camel case.